### ISSUE TITLE
Error running in WSL-2

### ISSUE THREAD

**Issue by rdar-lab** [Granulate/gprofiler#143]
I am using the latest docker version and getting an error when running.

The command I used:
```
docker run --name gprofiler --restart=always -d --pid=host --userns=host -v /var/run/docker.sock:/var/run/docker.sock -v /lib/modules:/lib/modules:ro -v /usr/src:/usr/src:ro --privileged granulate/gprofiler:latest -cu -
-token "XXXXXXXXX" --service-name "Service name"
```

The error I am getting:
```
Please run me in the init PID namespace! In Docker, make sure you pass '--pid=host'. In Kubernetes, add 'hostPID: true' in the Pod spec.
You can disable this check with --disable-pidns-check.
```

**Comment by Jongy** [comment]
This error suggests gProfiler isn't running in the init PID namespace. The common cause for it is forgetting to pass `--pid=host`/`hostPID: true`. We raise an hard error on this case - because in the standard case, it means gProfiler will only be able to profile processes running in *its* namespace (container), instead of system-wide. This is hardly ever what the user intended.

Now, in your case, you clearly are passing this flag. So I spent some time investigating WSL 2. Apparently, the shell I drop into when I type `wsl -d Ubuntu-20.04` on my test machine - runs in a non-init PID namespace. I'm not sure why WSL does it. This can be shown by running `ps -ef` in your WSL shell - you won't see any kernel threads (e.g `kthreadd`) because they run in the init PID namespace, just like you don't see kernel threads when you run `ps -ef` in a Docker container (unless `--pid=host` is passed).

As the error message suggests, if you pass `--disable-pidns-check` then gProfiler will skip this check. I checked gProfiler with it on my WSL 2 machine - it works just fine.
While proper support WSL 2 is not in our roadmap for now, it's obviously nice to support. I'll think of some way to improve the ["am I running in the init PID namespace" check](https://github.com/Granulate/gprofiler/blob/a376b98e729cbc310efe72e8a1a1c4b6caeeb2ae/gprofiler/utils.py#L478), so it works seamlessly also in WSL 2. Or, at least, explain it in the documentation.

Let me know if it works fine for you, so I can close this issue.
